include/unifex/async_scope.hpp

```diff
@@ -167,18 +167,35 @@ struct async_scope {
       just_from((Fun&&) fun));
   }
 
-  [[nodiscard]] auto cleanup() noexcept {
+  [[nodiscard]] auto complete() noexcept {
     return sequence(
-        just_from([this]() noexcept {
-          request_stop();
+        just_from([this] () noexcept {
+          end_of_scope();
         }),
         then(evt_.async_wait(), [this]() noexcept {
-          // make sure to synchronize with all the fetch_subs being done while
-          // operations complete
-          (void)opState_.load(std::memory_order_acquire);
+            // make sure to synchronize with all the fetch_subs being done while
+            // operations complete
+            (void)opState_.load(std::memory_order_acquire);
         }));
   }
 
+  [[nodiscard]] auto cleanup() noexcept {
+    return sequence(
+        just_from([this]() noexcept {
+          stopSource_.request_stop();
+        }),
+        complete());
+  }
```

Comment:
This is pretty nitpicky of me, but it's jumping out at me that `cleanup()` and `request_stop()` run `stopSource_.request_stop()` and `end_of_scope()` in different orders.  If this were my diff, I'd add a private `await_and_sync()`, (or something) that looks like this:
```
[[nodiscard]] auto await_and_sync() noexcept {
  return then(evt_.async_wait(), [this]() noexcept {
    // make sure to synchronize with all the fetch_subs being done while
    // operations complete
    (void)opState_.load(std::memory_order_acquire);
  });
}
```

Then I'd write `complete` and `cleanup` like this:
```
[[nodiscard]] auto complete() noexcept {
  return sequence(
      just_from([this]() noexcept {
        end_of_scope();
      }),
      await_and_sync());
 }

[[nodiscard]] auto cleanup() noexcept {
  return sequence(
      just_from([this]() noexcept {
        request_stop();
      }),
      await_and_sync());
}
```